### PR TITLE
netsniff-ng: add date format strings to --out

### PR DESCRIPTION
**Pull request by droberson**
This implements the functionality described in #158 

netsniff-ng -o output-%Y-%m-%d.pcap ### should output to output-2018-04-19.pcap

Signed-off-by: Daniel Roberson <daniel@planethacker.net>

### CODE REVIEW

**Comment by vkochan** [comment]
Hi Tobias!
It looks like what #158 is about. Of course needs to check with tcpdump behaviour, but in generally
it looks ok. Dont remember if there were some concerns in email thread which related to #158,
needs to check.

**Comment by droberson** [comment]
Requested style changes made.

**Comment by tklauser** [comment]
@droberson Thanks. Could you please squash the two commits?

@vkochan The `tcpdump` man page says "Savefiles will have the name specified by -w which should include a time format as defined by strftime(3)." and this PR uses strftime as well, so I guess it will match `tcpdump` behavior.

**Comment by tklauser** [comment]
Ok, I checked `tcpdump`'s implementation a bit more closely and it seems that it only expands the format specifiers if the `-G` option is given to rotate files. I think we should do something similar in netsniff-ng, i.e. only expand if the `-F/--interval` command line option is given. @droberson could you please update the implementation accordingly?

**Comment by droberson** [comment]
@tklauser Sure. I'll have a look this evening.

**Comment by tklauser** [comment]
@droberson I reconsidered what I wrote in https://github.com/netsniff-ng/netsniff-ng/pull/190#issuecomment-383107051 and now think your implementation is fine as is.

Since file rotation also applies if the `-o/--out` parameter is a directory (which `tcpdump` doesn't seem to support) and specifying a format there might be ambiguous. Also, we're not compatible to `tcpdump` in neither command line parameter nor in other implementation details. So I think it's fine to allow the users to give a time format string for the `-o/--out` parameter at any time - it might even be useful in case of dumping to a single file.

Long story short: now applied as e3e8eea4196688dd29a56efa940992f95d1ab141, thanks a lot!
